A lot of type errors

tsc version 4.9.5
lil-gui version 0.18.0

There now seems to be a lot more errors than it used to be regarding types:

<details>
<summary>Errors</summary>

```
node_modules/lil-gui/dist/lil-gui.esm.d.ts:305:5 - error TS2416: Property 'updateDisplay' in type 'BooleanController' is not assignable to the same property in base type 'Controller'.
  Type '() => BooleanController' is not assignable to type '() => this'.
    Type 'BooleanController' is not assignable to type 'this'.
      'BooleanController' is assignable to the constraint of type 'this', but 'this' could be instantiated with a different subtype of constraint 'BooleanController'.

305     updateDisplay(): BooleanController;
        ~~~~~~~~~~~~~

node_modules/lil-gui/dist/lil-gui.esm.d.ts:336:5 - error TS2416: Property 'reset' in type 'ColorController' is not assignable to the same property in base type 'Controller'.
  Type '() => ColorController' is not assignable to type '() => this'.
    Type 'ColorController' is not assignable to type 'this'.
      'ColorController' is assignable to the constraint of type 'this', but 'this' could be instantiated with a different subtype of constraint 'ColorController'.

336     reset(): ColorController;
        ~~~~~

node_modules/lil-gui/dist/lil-gui.esm.d.ts:340:5 - error TS2416: Property 'updateDisplay' in type 'ColorController' is not assignable to the same property in base type 'Controller'.
  Type '() => ColorController' is not assignable to type '() => this'.
    Type 'ColorController' is not assignable to type 'this'.
      'ColorController' is assignable to the constraint of type 'this', but 'this' could be instantiated with a different subtype of constraint 'ColorController'.

340     updateDisplay(): ColorController;
        ~~~~~~~~~~~~~

node_modules/lil-gui/dist/lil-gui.esm.d.ts:613:5 - error TS2416: Property 'decimals' in type 'NumberController' is not assignable to the same property in base type 'Controller'.
  Type '(decimals: any) => NumberController' is not assignable to type '(decimals: number) => this'.
    Type 'NumberController' is not assignable to type 'this'.
      'NumberController' is assignable to the constraint of type 'this', but 'this' could be instantiated with a different subtype of constraint 'NumberController'.

613     decimals(decimals: any): NumberController;
        ~~~~~~~~

node_modules/lil-gui/dist/lil-gui.esm.d.ts:615:5 - error TS2416: Property 'min' in type 'NumberController' is not assignable to the same property in base type 'Controller'.
  Type '(min: any) => NumberController' is not assignable to type '(min: number) => this'.
    Type 'NumberController' is not assignable to type 'this'.
      'NumberController' is assignable to the constraint of type 'this', but 'this' could be instantiated with a different subtype of constraint 'NumberController'.

615     min(min: any): NumberController;
        ~~~

node_modules/lil-gui/dist/lil-gui.esm.d.ts:617:5 - error TS2416: Property 'max' in type 'NumberController' is not assignable to the same property in base type 'Controller'.
  Type '(max: any) => NumberController' is not assignable to type '(max: number) => this'.
    Type 'NumberController' is not assignable to type 'this'.
      'NumberController' is assignable to the constraint of type 'this', but 'this' could be instantiated with a different subtype of constraint 'NumberController'.

617     max(max: any): NumberController;
        ~~~

node_modules/lil-gui/dist/lil-gui.esm.d.ts:619:5 - error TS2416: Property 'step' in type 'NumberController' is not assignable to the same property in base type 'Controller'.
  Type '(step: any, explicit?: boolean | undefined) => NumberController' is not assignable to type '(step: number) => this'.
    Type 'NumberController' is not assignable to type 'this'.
      'NumberController' is assignable to the constraint of type 'this', but 'this' could be instantiated with a different subtype of constraint 'NumberController'.

619     step(step: any, explicit?: boolean): NumberController;
        ~~~~

node_modules/lil-gui/dist/lil-gui.esm.d.ts:622:5 - error TS2416: Property 'updateDisplay' in type 'NumberController' is not assignable to the same property in base type 'Controller'.
  Type '() => NumberController' is not assignable to type '() => this'.
    Type 'NumberController' is not assignable to type 'this'.
      'NumberController' is assignable to the constraint of type 'this', but 'this' could be instantiated with a different subtype of constraint 'NumberController'.

622     updateDisplay(): NumberController;
        ~~~~~~~~~~~~~

node_modules/lil-gui/dist/lil-gui.esm.d.ts:649:5 - error TS2416: Property 'updateDisplay' in type 'OptionController' is not assignable to the same property in base type 'Controller'.
  Type '() => OptionController' is not assignable to type '() => this'.
    Type 'OptionController' is not assignable to type 'this'.
      'OptionController' is assignable to the constraint of type 'this', but 'this' could be instantiated with a different subtype of constraint 'OptionController'.

649     updateDisplay(): OptionController;
        ~~~~~~~~~~~~~

node_modules/lil-gui/dist/lil-gui.esm.d.ts:655:5 - error TS2416: Property 'updateDisplay' in type 'StringController' is not assignable to the same property in base type 'Controller'.
  Type '() => StringController' is not assignable to type '() => this'.
    Type 'StringController' is not assignable to type 'this'.
      'StringController' is assignable to the constraint of type 'this', but 'this' could be instantiated with a different subtype of constraint 'StringController'.

655     updateDisplay(): StringController;
        ~~~~~~~~~~~~~


Found 10 errors in the same file, starting at: node_modules/lil-gui/dist/lil-gui.esm.d.ts:305

```

</details>

Since my code hasn't changed (this was a dep bump from 0.17.0 to 0.18.0, and typescript from 4.9.4 to 4.9.5) I believe this has something to do with types

--- a/package.json
+++ b/package.json
@@ -55,7 +55,7 @@
     "rollup": "^1.17.0",
     "sass": "^1.22.10",
     "terser": "^4.3.8",
-    "typescript": "^4.5.2",
+    "typescript": "~4.5.5",
     "webfont": "^11.2.26"
   },
   "eslintIgnore": [

